ext/sockets/sockets.c

```diff
@@ -1883,11 +1884,19 @@ PHP_FUNCTION(socket_set_option)
 				RETURN_THROWS();
 			}
 
-			convert_to_long(l_onoff);
-			convert_to_long(l_linger);
+			zend_long zl_onoff = zval_try_get_long(l_onoff, &failed);
+			if (failed) {
+				zend_argument_type_error(4, "\"%s\" must be an int, %s given", l_onoff_key, zend_zval_value_name(l_onoff));
+				RETURN_THROWS();
+			}
+			zend_long zl_linger = zval_try_get_long(l_linger, &failed);
+			if (failed) {
+				zend_argument_type_error(4, "\"%s\" must be an int, %s given", l_linger_key, zend_zval_value_name(l_linger));
+				RETURN_THROWS();
+			}
 
-			lv.l_onoff = (unsigned short)Z_LVAL_P(l_onoff);
-			lv.l_linger = (unsigned short)Z_LVAL_P(l_linger);
+			lv.l_onoff = (unsigned short)zl_onoff;
+			lv.l_linger = (unsigned short)zl_linger;
```

Comment:
Probably the zend_long should be checked to see if it fits in an unsigned short before.

Reply:
Already happens in the lines above.